### PR TITLE
cml-pr name branch

### LINKED ISSUE #551: cml-pr convention name seems to be breaking GH's branches page
seems to be perfectly fine in BB and GL.

image

### PR DESCRIPTION
**Pull request by DavidGOrtega**
GH gives branch names with all the ref ```/refs/heads/branch``` using branch name like this seems possible through their api but is breaking later on their UI
 
- Extracts the last part of the branch name
 
closes #551

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
```console
TypeError: Cannot read property 'split' of undefined
    at branch_name (/usr/local/lib/node_modules/@dvcorg/cml/src/drivers/github.js:22:24)
    at Github.get branch [as branch] (/usr/local/lib/node_modules/@dvcorg/cml/src/drivers/github.js:298:12)
    at CML.branch (/usr/local/lib/node_modules/@dvcorg/cml/src/cml.js:88:13)
    at CML.pr_create (/usr/local/lib/node_modules/@dvcorg/cml/src/cml.js:267:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async run (/usr/local/lib/node_modules/@dvcorg/cml/bin/cml-pr.js:15:10)
```

Running locally in the absence of the `GITHUB_REF` environment variable. Should we cover this edge case?

**Comment by 0x2b3bfa0** [comment]
Branches with slashes in their names, like `experiment/break-everything` won't work on GitHub and, if they did, would produce inconsistent results with other drivers. Admittedly, I did it on purpose. 😈  

```console
GITHUB_REF=experiment/break-everything cml-pr ···
```
```javascript
{
  status: 422,
  headers: {···},
  request: {
    method: 'POST',
    url: 'https://api.github.com/repos/···/···/pulls',
    headers: {···},
    body: '{"head":"break-everything-cml-pr-···","base":"break-everything","title":"CML PR for break-everything 1a6ba96f","body":"\\nAutomated commits for https://github.com/···/···/commit/··· created by CML.\\n  "}',
    request: {···}
  },
  errors: [ { resource: 'PullRequest', field: 'base', code: 'invalid' } ],
  documentation_url: 'https://docs.github.com/rest/reference/pulls#create-a-pull-request'
}
```

File: src/drivers/github.js [hunk 1]
```diff
@@ -290,7 +297,7 @@ class Github {
   }
 
   get branch() {
-    return GITHUB_REF;
+    return branch_name(GITHUB_REF);
```
Comment (0x2b3bfa0) [inline src/drivers/github.js:300]:
```suggestion
    return GITHUB_REF;
```

It doesn't contain the `refs/heads` prefix, so removing any path components will break on branch names with slashes.

Comment (DavidGOrtega) [reply src/drivers/github.js:300]: @0x2b3bfa0 true. So we have to do it the hard way

Comment (DavidGOrtega) [reply src/drivers/github.js:299]: GITHUB_REF contains refs/heads

Comment (DavidGOrtega) [reply src/drivers/github.js:299]: https://stackoverflow.com/questions/58033366/how-to-get-current-branch-within-github-actions

Comment (0x2b3bfa0) [reply src/drivers/github.js:299]: 🤦🏼  My bad: tested in the wrong environment, sorry.

**Review by 0x2b3bfa0 — APPROVED** [review]
Looks good to me! Nevertheless, the original error still feels puzzling.

**Comment by casperdcl** [comment]
> Base branches with slashes in their names, like `experiment/break-everything` won't work on GitHub

They should? Sounds like a big

**Comment by DavidGOrtega** [comment]
@casperdcl @0x2b3bfa0  only happens apparently with refs/heads/

**Comment by 0x2b3bfa0** [comment]
> _They should? Sounds like a bug_

They should. [Period.](https://git-scm.com/docs/git-check-ref-format) 😉

**Comment by 0x2b3bfa0** [comment]
> _only happens apparently with refs/heads_

Apparently, because, I can't reproduce it with a minimal example. It's something abut a repository.

File: src/drivers/github.js [hunk 2]
```diff
@@ -18,6 +18,12 @@ const {
   GITHUB_EVENT_NAME
 } = process.env;
 
+const branch_name = (branch) => {
+  if (!branch) return;
+
+  return branch.replace('refs/heads/', '');
```
Comment (casperdcl) [inline src/drivers/github.js:24]: does we need logic for `refs/tags/` too?

Comment (0x2b3bfa0) [reply src/drivers/github.js:24]: 👁️  💯  Probably yes, users may trigger the workflow on tag events.

Comment (casperdcl) [reply src/drivers/github.js:24]:
```suggestion
  return branch.replace(/refs\/(head|tag)s\//, '');
```

Comment (DavidGOrtega) [reply src/drivers/github.js:24]: For security I guess so, I dont see when that would happen 🤔 but definitely is secure

**Review by casperdcl — APPROVED** [review]

**Pull request closed without merging**